src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/IPAddressUtil.cs

```diff
@@ -20,7 +20,9 @@ public static bool IsMulticast(IPAddress address)
             }
             else
             {
-                byte firstByte = address.GetAddressBytes()[0];
+#pragma warning disable CS0618 // using Obsolete Address API because it's the more efficient option in this case
+                byte firstByte = (byte)address.Address;
+#pragma warning restore CS0618
                 return firstByte >= 224 && firstByte <= 239;
```

Comment:
```suggestion
                return firstByte is >= 224 and <= 239;
```
Once roslyn optimizes this (cf. https://github.com/dotnet/roslyn/issues/60534), codegen improvement comes for free.

Reply:
I understand these are different constructs as far as Roslyn is concerned, but it's an unfortunate pit of failure if these otherwise identical expressions end up being optimized differently. I'd really like for us to not have to change all such range checks everywhere in order for the right things to happen.